datadog_lambda/tracing.py

```diff
@@ -272,8 +307,9 @@ def extract_context_from_sqs_or_sns_event_or_context(event, lambda_context):
                         logger.debug(
                             "Failed to extract Step Functions context from SQS/SNS event."
                         )
-
-                return propagator.extract(dd_data)
+                if is_sqs:
+                    return _extract_context(dd_data, "sqs", arn)
+                return _extract_context(dd_data, "sns", sns_record.get("TopicArn", ""))
```

Comment:
At this point, I know it's pretty impossible for `sns_record` not to be defined. But there is still the chance of it, especially in the future if someone makes changes. Here's what I'd recommend to protect against that.

Move the sns arn extraction to inside the if block where `sns_record` is defined. Then here, all you'll need to do is

```python
return _extract_context(dd_data, "sqs" if is_sqs else "sns", arn)
```